dist/angular-dynamic-layout.js

```diff
@@ -153,20 +160,24 @@
 
   angular
     .module('dynamicLayout')
-    .directive('layoutOnLoad', ['$rootScope', layoutOnLoad]);
+    .directive('layoutOnLoad', ['$rootScope', '$timeout', layoutOnLoad]);
 
   /*
    * Directive on images to layout after each load
    */
-  function layoutOnLoad($rootScope) {
+  function layoutOnLoad($rootScope, $timeout) {
 
     return {
-        restrict: 'A',
-        link: function(scope, element) {
-          element.bind('load error', function() {
-            $rootScope.$broadcast('layout');
+      restrict: 'A',
+      link: function(scope, element) {
+        element.bind('load error', function() {
+          var timeoutId;
```

Comment:
If you set it here it will be reinitialized everytime the callback is executed, just declare it outside the callback and it should work so [here](https://github.com/tristanguigue/angular-dynamic-layout/pull/35/files#diff-22a5340a769d53243fa3dec01fcd55a0R173)